meetings/2024-01-17.md

```diff
@@ -0,0 +1,161 @@
+* MdJ: That'd be perhaps in future spec?
+* SC: Current spec says.. category of contraints, one imposed by the server, expectation that container should behave.. Anything beyond is open, spec allows the case PAC was mentioning. There's also the possibility to advertise those contraints. Existing servers, if they want to continue doing what they're doing might want to get into that habit. 
+* SC: There's a similar consideration raised in Solid Protocol, whether a read-only Solid server should be conforming. Normally we think read-write. Wehther a conforming server not implementing those methods.. Some servers might want to do the mminimal thing. That's fine. 
+* MdJ: "Advertising its chocolateness" (TM).
+* ML: So this constraint would be know at run-time and not when they're choosing their pod. So then I get a pod but not might as expected.
+* eP: Separate topic perhaps how people can choose. Who is setting the constraints. It wouldn't be arbitrary constraints. How we can educate users to picking a storage.
+* SC: Server could pass the test suite. When it's deployed it could completely reject requests including multiple subject. I don't think test suite will catch that. Comes down to whether URI owner allows that. 
+
+
+### Demo
+#### Solid Data Modules Intro
+URL: https://github.com/solid-contrib/data-modules
+
+Proposed & presented by: Michiel
+
+* MdJ: Slides: https://github.com/michielbdejong/presentations/blob/main/sdm-cg-17-jan.pdf
+* MdJ: Solid Data Modules uses the platform for data interopearbility. I'd call "'small' Solid = RWW". As a power user there can be data browsers. mashlib is one. Also from modula, ontolo. Kind of like SolidOS where any data has a particular shape can have a view. I also see a bigger vision "'big' Solid = data pivot". It is a datastore. The GUI running on top of it. We need a much stricter spec for pod as a data pirvot. App A writes to Solid Pod and read by App B. For one power user, usefullness comes if it can be reused by App B. For that to open apps need to know what to expect from the pod, when it will fail, or discoverably capabilities for minimal reliable working thing. Data application rights need to be available in an understandable way. but then the choice of ontology and data shape matters. The business logic around that need sto be understood. need server behaviour to be strict - can't have chocolate servers - have data conventions, which ontologies to use - even if we care a WebID Profile it owuld have x:name but the app is using y:name. No good data conventions and also access control and they go hand in hand. The activities in Data Modules, we document and write translations between formats, reusable code in JavaScript, which handles the translation for you. This is one of the sites where we document data conventions. For example, bookmarking is in different vocabs. It'd be nice to make that compatible. Some apps use x and other y without knowing the other exists. SDM also abstracts them.
```

Comment:
```suggestion
* MdJ: Solid Data Modules uses the platform for data interoperability. I'd call "'small' Solid = RWW". As a power user, there can be data browsers. `mashlib` is one. Also from modula, `ontolo`. Kind of like SolidOS, where any data has a particular shape, can have a view. I also see a bigger vision "'big' Solid = data pivot". It is a datastore. The GUI running on top of it. We need a much stricter spec for pod as a data pivot. App A writes to Solid Pod, read by App B. For one power user, usefulness comes if App A's data can be reused by App B. For that to be open, apps need to know what to expect from the pod, when it will fail, or have discoverable capabilities for minimal reliable working thing. Data application rights need to be available in an understandable way, but then the choice of ontology and data shape matters. The business logic around that needs to be understood. need server behaviour to be strict — can't have chocolate servers — have data conventions, which ontologies to use — even if we share a WebID Profile, it might have `x:name`, but the app is using `y:name`. No good data conventions and also access control and they go hand in hand. The activities in Data Modules, we document and write translations between formats, reusable code in JavaScript, which handles the translation for you. This is one of the sites where we document data conventions. For example, bookmarking is in different vocabs. It'd be nice to make that compatible. Some apps use `x` and other `y` without knowing the other exists. SDM also abstracts them.
```